Eplant/tutorial.md

```diff
@@ -0,0 +1,227 @@
+    geneticElement={gene}
+    activeData={activeData}
+    dispatch={dispatch}
+/>
+```
+
+
+### useViewData
+The `useViewDatai` function is a custom React hook designed to manage and load view data in the application. It leverages TypeScript generics to handle different types of view data, state, and actions.
+
+See `Eplant/View/viewData.ts` for more details.
+
+
+
+## View Interface
+Please see the `View` interface in `View \index.ts`
+This is the interface we use for most of the `views`
+
+The View interface is a TypeScript interface that defines the structure and functionality expected from a view component in a React application.
+
+
+### getInitialState:
+
+An optional method that returns the initial state of the view. If not defined, the state defaults to null.
+This method initializes the state when the view is first rendered.
+
+
+### getInitialData:
+
+A required method that fetches the data needed for the view based on a given gene and provides loading feedback.
+
+### reducer:
+
+An optional reducer function used to handle actions and update the state. This is useful for managing state changes based on different actions.
+
+
+### actions:
+
+An optional array of actions that can be shown in the view’s options menu. Each action allows interaction with the view.
+
+
+### component:
+
+A required React component that renders the view.
+Takes props of type ViewProps<Data, State, Action>.
+
+
+### icon:
+
+An optional React component that returns an icon representing the view. This is used in the gene info viewer to visually identify the view.
+
+### name:
+
+A required string that represents the name of the view. This is used to identify the view in various contexts.
+
+### id:
+
+A required string that provides a unique identifier for the view.
+
+
+### description:
+
+An optional string that provides a description of the view.
+
+### thumbnail:
+
+An optional string that represents the URL or path to a thumbnail image for the view.
+
+### citation:
+
+An optional React component that renders citations related to the gene. This helps in displaying references or additional information.
+Takes props which may include state, activeData, and gene.
+
+### header:
+
+An optional function that returns the title of the view's tab.
+Takes props of type `ViewProps<Data, State, Action>`.
+
+
+## state
+`state` (`Eplant/state/index.tsx`) is a React module that manages state using `Jotai` and provides various hooks and utility functions for managing application state in the application, which is similar to `Redux` and `React Hooks`.
+
+### Atoms: 
+Basic units of state. An atom holds a piece of state that can be read from and written to.
+### Hooks: 
+useAtom is used to interact with atoms, allowing components to access and update state.
+### Simplicity:
+Minimal boilerplate and API surface, making it easy to understand and use.
+### Reactivity: 
+Components automatically re-render when atom state changes.
+
+## When you create a new `view`
```

Comment:
@Yukthiw can you review this section?

Reply:
@y330 if you can take a look as well that would be great. Thanks!

Reply:
Looks good to me!